### PR TITLE
Update docs on OData exceptions

### PR DESCRIPTION
**Pull request by MatKuhr**
## Context

Update OData exception class names according to latest changes.

### CODE REVIEW

File: docs/java/features/odata/type-safe-client-odata-v2.mdx [hunk 1]
```diff
@@ -413,20 +413,24 @@ try { ... }
 }
 ```
 
-This handling is the most generic, handling all possible failures. For more specific information there are dedicated exceptions inheriting from `ODataException`. Please tend to the [documentation](https://help.sap.com/doc/b579bf8578954412aea2b458e8452201/1.0/en-US/com/sap/cloud/sdk/datamodel/odata/client/exception/ODataException.html) for all the exception types.
+This handling is the most generic, handling all possible failures.
+For more specific information there are dedicated exceptions inheriting from `ODataException`. Please tend to the [documentation](https://help.sap.com/doc/b579bf8578954412aea2b458e8452201/1.0/en-US/com/sap/cloud/sdk/datamodel/odata/client/exception/ODataException.html) for all the exception types.
```
Comment (Emdee89) [inline docs/java/features/odata/type-safe-client-odata-v2.mdx:417]: Does this Javadoc link refer always to the latest version?

**Comment by Emdee89** [comment]
Proposal: Add one concrete example that demonstrates treating an error with a concrete exception class. Currently the documentation refers to the Javadoc. An example might be more insightful for less experienced developers.

File: docs/java/features/odata/type-safe-client-odata-v2.mdx (hunk 1 above)
Comment (MatKuhr) [reply docs/java/features/odata/type-safe-client-odata-v2.mdx:417]: It should, yes. But there is no guarantee. Currently it points to 3.30 but I think 3.31 wasn't uploaded yet. I haven't checked.

**Comment by MatKuhr** [comment]
> Proposal: Add one concrete example that demonstrates treating an error with a concrete exception class. Currently the documentation refers to the Javadoc. An example might be more insightful for less experienced developers.

But that is already in place. The example code shows how to get the OData error code specifically dealing with the `ODataServiceErrorException`. IMO this is a concrete example that you could copy paste into your production code. So what kind of example would you add here?

**Review by Emdee89 — APPROVED** [review]

**Comment by Emdee89** [comment]
> > Proposal: Add one concrete example that demonstrates treating an error with a concrete exception class. Currently the documentation refers to the Javadoc. An example might be more insightful for less experienced developers.
> 
> But that is already in place. The example code shows how to get the OData error code specifically dealing with the `ODataServiceErrorException`. IMO this is a concrete example that you could copy paste into your production code. So what kind of example would you add here?

I had accessing response headers and response code in mind. But grabbing the OData error is also a meaningful example.

**Pull request merged**